swiftide-indexing/src/transformers/file_to_context_tree_sitter.rs

```diff
@@ -0,0 +1,64 @@
+//! Add file-level context to the metadata of a node containing code using the tree-sitter parser.
+use anyhow::Result;
+use async_trait::async_trait;
+use derive_builder::Builder;
+
+use swiftide_core::indexing::Node;
+use swiftide_core::Transformer;
+
+use swiftide_integrations::treesitter::{CodeOutliner, SupportedLanguages};
+
+pub const NAME: &str = "Context (code)";
+
+/// `FileToContextTreeSitter` adds a "Context (Code)" field to the metadata of a node that contains
+/// a summary of the code in the node. It uses the tree-sitter parser to parse the code and
+/// remove any information that is less relevant for tasks that consider the file as a whole.
+#[derive(Debug, Clone, Builder)]
+#[builder(pattern = "owned", setter(into, strip_option))]
+pub struct FileToContextTreeSitter {
```

Comment:
I'd suggest to rename this to `OutlineCode`, with some docs on that it does does that and is intended to be used before chunking to potentially add to future chunks as extra context.